src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.hpp

```diff
@@ -51,12 +51,23 @@ struct BrgemmKernelConfig : public snippets::KernelExecutorBase::GenericConfig {
 #endif
 
 private:
+    struct StaticParams {
+        StaticParams(const element::Type& in0_dtype, const element::Type& in1_dtype, float beta,
+                            bool is_with_amx, bool is_with_comp);
+        const dnnl_data_type_t dt_in0 {dnnl_f32}, dt_in1 {dnnl_f32};
+        const float beta {0};
+        const bool is_with_amx {false};
+        const bool is_with_comp {false};
+        const dnnl::impl::cpu::x64::cpu_isa_t isa {dnnl::impl::cpu::x64::isa_undef};
+        const size_t hash{0};
+        bool operator==(const StaticParams& rhs) const;
+        bool operator!=(const StaticParams& rhs) const {return !(*this == rhs);}
+#ifdef SNIPPETS_DEBUG_CAPS
+        std::string to_string() const;
+#endif
+    };
     size_t compute_hash() const;
-    dnnl_data_type_t m_dt_in0 {dnnl_f32}, m_dt_in1 {dnnl_f32};
-    bool m_is_with_amx {false};
-    bool m_is_with_comp {false};
-    float m_beta {0};
-    dnnl::impl::cpu::x64::cpu_isa_t m_isa {dnnl::impl::cpu::x64::isa_undef};
+    std::shared_ptr<StaticParams> m_static_params;
```

Comment:
Yep, we really need this pointer.
The whole idea behind introducing the `StaticParams` is to avoid copying of shape-agnostic parameters while creating specialized kernel executors. So if we just need to create brgemm with new shapes, we don't need to copy isa/types/beta etc.
As for the segfault - it's a valid point. Let's delete the default constructor and make the `m_static_params` const. This way can guarantee that `m_static_params` is always valid.

Reply:
Got it, thank you for the explanation!